site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -247,35 +515,101 @@ def transform_custom_fields(self):
             if x.get("legacy_name")
         ]
 
-        custom_fields["cern:accelerators"] = accelerators
-        custom_fields["cern:experiments"] = experiments
+        imprint = self._validate_imprint()
+
+        if accelerators:
+            custom_fields["cern:accelerators"] = accelerators
+        if experiments:
+            custom_fields["cern:experiments"] = experiments
+        if imprint and imprint.get("place"):
+            custom_fields["imprint:imprint"] = {"place": imprint.get("place")}
+
+        imprint_fields = custom_fields.get("imprint:imprint", {})
+
+        isbns = self.inspire_metadata.get("isbns", [])
+        online_isbns = []
+        for isbn in isbns:
+            value = isbn.get("value")
+            valid_isbn = normalize_isbn(value)
+            if not valid_isbn:
+                self.metadata_errors.append(f"Invalid ISBN '{value}'.")
+            else:
+                if isbn.get("medium") == "online":
+                    online_isbns.append(valid_isbn)
```

Comment:
this will be true for thesis, but not so much for books. Can you open a ticket to revisit this in the future?